tests/integration/settings/project-settings/general/project-name-settings.test.ts

```diff
@@ -0,0 +1,118 @@
+                loadingProject: Promise.resolve(Projects.demoProject),
+            },
+        });
+
+        expect(screen.queryByText("General")).toBeInTheDocument();
+        expect(screen.queryByText("Project Name")).toBeInTheDocument();
+        const projectRenameInput = screen.getByLabelText("Project Name");
+        expect(projectRenameInput).toBeInTheDocument();
+        await waitFor(() => {
+            expect(projectRenameInput).toHaveValue(Projects.demoProject.name);
+        });
+        const renameButton = screen.getByRole("button", { name: "Rename" });
+        expect(renameButton).toBeInTheDocument();
+    });
+
+    test("When the input field is empty or the project name is the same, then the button should not change the name", async () => {
+        const mockCall = vi.spyOn(backendService, "getProjectById");
+
+        render(ProjectNameSettings, {
+            target: document.body,
+            props: {
+                projectId: Projects.demoProject.id,
+                loadingProject: Promise.resolve(Projects.demoProject),
+            },
+        });
+
+        const projectRenameInput = screen.getByLabelText("Project Name");
+        const renameButton = screen.getByRole("button", { name: "Rename" });
+
+        await userEvent.clear(projectRenameInput);
+        await userEvent.type(projectRenameInput, " ");
+        await userEvent.click(renameButton);
+        await userEvent.clear(projectRenameInput);
+        await userEvent.type(projectRenameInput, "Demo Project");
+        await userEvent.click(renameButton);
+        expect(mockCall).not.toHaveBeenCalled();
+    });
+
+    test("When the input field is filled with a differing name, then the button should change the name", async () => {
+        const mockCall = mockApiCall("updateProject", {
+            id: "0",
+            name: "New Project Name",
+            status: ProjectStatus.ACTIVE,
+            currentStage: 0n,
+            maxStage: 1n,
+            settings: undefined,
+        });
+
+        render(ProjectNameSettings, {
+            props: {
+                projectId: Projects.demoProject.id,
+                loadingProject: Promise.resolve(Projects.demoProject),
+            },
+        });
+
+        const projectRenameInput = screen.getByLabelText("Project Name");
+        const renameButton = screen.getByRole("button", { name: "Rename" });
+
+        await userEvent.clear(projectRenameInput);
+        await userEvent.type(projectRenameInput, "New Project Name");
+        await userEvent.click(renameButton);
+        expect(mockCall).toHaveBeenCalled();
+    });
+
+    test("When a failed API call to get the project is made, then an error message should be shown", async () => {
+        const failedPromise = new Promise<Project>((_, reject) => {
+            setTimeout(() => reject("Error while loading project"), 0);
+        });
+        vi.spyOn(console, "error").mockImplementation(() => {});
```

Comment:
Is there a reason why we use a timeout with 0 ms? These lines can be simplified. Also, let's not mock the console logging.
```suggestion
        const failedPromise = Promise.reject(new Error("Error while loading project"));
```

Reply:
Thank you for the hint. Now it makes sense where the error regarding to this ``` ?.response.then(() => {``` comes from. Totally makes sense to mock all calls that are called. Will do that in the future.

Reply:
Thank you for the hint. Now it makes sense where the error regarding to this  ?.response.then(() => { comes from. Totally makes sense to mock all calls that are called. Will do that in the future.